Watchman.Discord/Areas/Protection/RequestParsers/RequestParser.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Devscord.DiscordFramework.Commons.Exceptions;
+using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Watchman.Discord.Areas.Commons;
+
+namespace Watchman.Discord.Areas.Protection.Services
```

Comment:
czemu RequestParser który z tego co widze powinien być uniwersalny
jest ograniczony do używania go w Areas.Protection?

Reply:
Nie jestem autorem tej klasy a nie poprawiłem, bo nie wiem gdzie wsadzić żeby było dobrze

Reply:
z tego co widze po tym co code review mi pokazuje, to pojawiła się dopiero teraz

ale wracając -> unikamy logiki związanej z DiscordRequest bo wszystko staramy się przepiąć na IBotCommand

Reply:
> ale wracając -> unikamy logiki związanej z DiscordRequest bo wszystko staramy się przepiąć na IBotCommand
ok

> z tego co widze po tym co code review mi pokazuje, to pojawiła się dopiero teraz
a) na początku była klasa MuteRequestParser
b) wtedy robiłem źle - dorobiłem WarnRequestParser a że miały części wspólne to wspólną część wyniosłem do RequestParser
c) usunąłem WarnRequestParser bo jednak robię IBotCommand

Reply:
ogólnie wynika to z tego że projekt się rozwija
stopniowo zauważamy że pewne rzeczy są problematyczne
więc je poprawiamy tak, żeby używanie ich było bardziej intuicyjne, bezpieczne i proste
jednak z czasem pojawiają się problemy których nie przewidzieliśmy wcześniej
i musimy robić kolejne zmiany
czasami są to delikatne poprawki
a czasami solidne przebudowy mechanizmów prawie że od podstaw

przykładowo workflow, antyspam albo te komendy
to elementy które były u nas przebudowywane kilka razy prawie że od zera 
więc w pewnych miejscach mogą być pozostałości po starych wersjach 

jak chcesz się na czymś wzorować, to najlepiej nie wzoruj się na jednym konkretnym miejscu
tylko przejrzyj wiele, żeby mieć szerszy pogląd 

tak samo zdarza się czasami że ktoś zrobi błąd, bo o czymś zapomni
a druga osoba na etapie review tego błędu nie zauważy

żeby review było w pełni błędo-odporne musiałoby to robić pewnie z 20 osób
aktualnie to są 2 osoby w przypadku nowych osób
zdarzają się sytuacje (i dawniej tak było) że całe review robi jedna osoba

jeśli zmiany są małe, bo są na 2-3 pliki, to łatwo zauważyć większość błędów
gorzej jeśli pull request ma w okolicy 50 plików -> tutaj już łatwo coś przeoczyć

Reply:
RequestParsera już z nami nie ma